File: Example/iOS-Objective-C/PaintingViewController.m
//
//  PaintingViewController.m
//  iOS-Objective-C
//
//  Created by Felix Herrmann on 14.10.21.
//

#import "PaintingViewController.h"
#import "iOS_Objective_C-Swift.h"


@interface PaintingViewController ()

@property UIImageView *imageView;

@end


@implementation PaintingViewController


#pragma mark - Initializers

- (instancetype)initWithPainting:(Painting *)painting {
    if(self = [super init]) {
        self.painting = painting;
    }
    
    return self;
}


#pragma mark - ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    
    [self setupView];
}


#pragma mark - Setup

- (void)setupView {
    self.imageView = [[UIImageView alloc] initWithImage:self.painting.image];
    self.imageView.contentMode = UIViewContentModeScaleAspectFill;
    self.imageView.clipsToBounds = YES;
    self.imageView.translatesAutoresizingMaskIntoConstraints = NO;
    
    [self.view addSubview:self.imageView];
    
    [NSLayoutConstraint activateConstraints:@[
        [self.imageView.leadingAnchor constraintEqualToAnchor:self.view.readableContentGuide.leadingAnchor constant:20],
        [self.imageView.topAnchor constraintEqualToAnchor:self.view.readableContentGuide.topAnchor constant:160],
        [self.imageView.trailingAnchor constraintEqualToAnchor:self.view.readableContentGuide.trailingAnchor constant:-20],
        [self.imageView.bottomAnchor constraintEqualToAnchor:self.view.readableContentGuide.bottomAnchor constant:-140]
    ]];
}

@end

File: Example/iOS/PaintingViewController.swift
//
//  PaintingViewController.swift
//  iOS
//
//  Created by Felix Herrmann on 14.10.21.
//

import UIKit

final class PaintingViewController: UIViewController {
    
    
    // MARK: - Properties
    
    let painting: Painting
    
    private let imageView = UIImageView()
    
    
    // MARK: - Initializers
    
    init(painting: Painting) {
        self.painting = painting
        super.init(nibName: nil, bundle: nil)
    }
    
    required init?(coder: NSCoder) {
        fatalError("init(coder:) has not been implemented")
    }
    
    
    // MARK: - ViewController
    
    override func viewDidLoad() {
        super.viewDidLoad()
        
        setupView()
    }
    
    
    // MARK: - Setup
    
    private func setupView() {
        imageView.image = painting.image
        imageView.contentMode = .scaleAspectFill
        imageView.clipsToBounds = true
        imageView.translatesAutoresizingMaskIntoConstraints = false
        
        view.addSubview(imageView)
        
        NSLayoutConstraint.activate([
            imageView.leadingAnchor.constraint(equalTo: view.readableContentGuide.leadingAnchor, constant: 20),
            imageView.topAnchor.constraint(equalTo: view.readableContentGuide.topAnchor, constant: 160),
            imageView.trailingAnchor.constraint(equalTo: view.readableContentGuide.trailingAnchor, constant: -20),
            imageView.bottomAnchor.constraint(equalTo: view.readableContentGuide.bottomAnchor, constant: -140)
        ])
    }
}
